source/magpie/dataframe.d

```diff
@@ -187,25 +188,47 @@ public:
                     maxGap = (maxGap > lenCol)? maxGap: lenCol;
                 }
 
-                static foreach(j; 0 .. RowType.length)
+                static if(isHomogeneousType)
```

Comment:
create an auxiliary function that does something like 

```
void homogenousDispatch(alias func)
{
     static if(isHomogeneousType)
         func(dataIndex);
    else
        static foreach(j; 0 .. RowType.length)
              if(j == dataIndex)
                     func(dataIndex);
}
```

and  the call it like

```
homogenousDispatch!(
    (j)
    {
          size_t maxsize1 = 0, maxsize2 = 0;
          // ...
    });
```

Ditto throughout.

If you've got a better name for it please use it instead.

Reply:
Figured out dispatcher long at last :sweat_smile: